### PR TITLE
Minimal tracing support

### PR DESCRIPTION
**Pull request by andyyu2004**
Add client spans and support `request-middleware::ClientWithMiddleware`.

### CODE REVIEW

File: reflectapi/Cargo.toml [hunk 1]
```diff
@@ -53,6 +53,7 @@ check_keyword = { version = "0.2.0", optional = true }
 
 # optional 3rd party dependencies for client runtime
 reqwest = { version = "0.12", optional = true }
+reqwest-middleware = { version = "0.3", optional = true }
```
Comment (hardbyte) [inline reflectapi/Cargo.toml:56]: Don't you have to add this as a feature as well?

I assume the clients will have to depend on `reqwest-tracing`, and create their custom reqwest client - might be worth updating the demo to show how to use it.

File: reflectapi/src/rt.rs [hunk 2]
```diff
@@ -188,23 +191,28 @@ impl Client for reqwest::Client {
...
+#[cfg(feature = "reqwest-middleware")]
+impl Client for reqwest_middleware::ClientWithMiddleware {
+    type Error = reqwest_middleware::Error;
+
+    async fn request(
+        &self,
+        path: Url,
+        body: bytes::Bytes,
+        headers: http::HeaderMap,
+    ) -> Result<(http::StatusCode, bytes::Bytes), Self::Error> {
```
Comment (hardbyte) [inline reflectapi/src/rt.rs:212]: Nice

File: reflectapi-cli/src/main.rs [hunk 3]
```diff
@@ -86,7 +86,9 @@ fn generate(
 
     let config = reflectapi::codegen::Config {
         format: true,
-        typecheck: true,
+        // Rust typechecking is a bit limited.
+        typecheck: matches!(language, Language::Typescript),
+        tracing: true,
```
Comment (hardbyte) [inline reflectapi-cli/src/main.rs:91]: Nice touch making the `#instrument` annotations opt in.

**Review by hardbyte — APPROVED** [review]

File: reflectapi/Cargo.toml (hunk 1 above)
Comment (andyyu2004) [reply reflectapi/Cargo.toml:56]: Yep pretty much. Can do. I think it implicitly creates a feature under the same name. Not sure if that method is deprecated or not.

File: reflectapi/src/codegen/rust-dependency-stubs/http.rs [hunk 4]
```diff
@@ -17,3 +18,56 @@ impl StatusCode {
...
+    pub fn insert(&mut self, _name: HeaderName, _value: T) {
+        unimplemented!()
+    }
+}
+
+pub struct HeaderName(());
+
+impl HeaderName {
+    pub fn from_bytes(_bytes: &[u8]) -> Result<Self, InvalidHeaderName> {
+        unimplemented!()
```
Comment (avkonst) [inline reflectapi/src/codegen/rust-dependency-stubs/http.rs:41]: it is not real code? only stubs?

Comment (andyyu2004) [reply reflectapi/src/codegen/rust-dependency-stubs/http.rs:41]: Yeah, it's a bit of a hack to do typechecking to ensure the generated rust code is correct for tests at least. I don't mind removing it but it's pretty helpful.

**Pull request merged**